cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/artifact/RemoteArtifactRepository.java

```diff
@@ -0,0 +1,204 @@
+  public List<ArtifactSummary> getArtifactSummaries(ArtifactRange range, int limit,
+                                                    ArtifactSortOrder order) throws Exception {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public List<ApplicationClassSummary> getApplicationClasses(NamespaceId namespace,
+                                                             boolean includeSystem) throws IOException {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public List<ApplicationClassInfo> getApplicationClasses(NamespaceId namespace, String className) throws IOException {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public SortedMap<ArtifactDescriptor, Set<PluginClass>>
+  getPlugins(NamespaceId namespace,
+             Id.Artifact artifactId) throws IOException, ArtifactNotFoundException {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public SortedMap<ArtifactDescriptor, Set<PluginClass>>
+  getPlugins(NamespaceId namespace, Id.Artifact artifactId,
+             String pluginType) throws IOException, ArtifactNotFoundException {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public SortedMap<ArtifactDescriptor, PluginClass>
+  getPlugins(NamespaceId namespace, Id.Artifact artifactId,
+             String pluginType, String pluginName,
+             Predicate<ArtifactId> pluginPredicate, int limit,
+             ArtifactSortOrder order) throws IOException, PluginNotExistsException, ArtifactNotFoundException {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public Map.Entry<ArtifactDescriptor, PluginClass>
+  findPlugin(NamespaceId namespace, ArtifactRange artifactRange,
+             String pluginType, String pluginName,
+             PluginSelector selector) throws IOException, PluginNotExistsException, ArtifactNotFoundException {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public ArtifactDetail addArtifact(Id.Artifact artifactId, File artifactFile) throws Exception {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public ArtifactDetail addArtifact(Id.Artifact artifactId, File artifactFile,
+                                    @Nullable Set<ArtifactRange> parentArtifacts,
+                                    @Nullable Set<PluginClass> additionalPlugins) throws Exception {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public ArtifactDetail addArtifact(Id.Artifact artifactId, File artifactFile,
+                                    @Nullable Set<ArtifactRange> parentArtifacts,
+                                    @Nullable Set<PluginClass> additionalPlugins,
+                                    Map<String, String> properties) throws Exception {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void writeArtifactProperties(Id.Artifact artifactId, Map<String, String> properties) throws Exception {
+
```

Comment:
Shouldn't all these unimplemented methods throwing the UnsupportedOperationException?

Reply:
yes...not sure how I missed that. Its fixed now